frontend/public/components/dashboard/details-card/detail-item.tsx

```diff
@@ -12,8 +15,28 @@ export const DetailItem: React.FC<DetailItemProps> = React.memo(({ title, value,
   </React.Fragment>
 ));
 
+
+export const DetailItemLink: React.FC<DetailItemLinkProps> = React.memo(({ title, value, link, isLoading }) => (
+  <React.Fragment>
+    <dt className="co-details-card__item-title">{title}</dt>
+    <dd className={classNames('co-details-card__item-value', {'text-secondary': !value})}>
+      {isLoading ? <LoadingInline /> : (
+        value ? <ExternalLink href={link} text={value} /> : 'Unavailable'
```

Comment:
systemName being tested to see if it is undefined. Since systemName and systemAddress and queried from the same object and at the same depth. @afreen23